source/write-operations/insert-documents.txt

```diff
@@ -0,0 +1,208 @@
+.. :ref:`<java-rs-getting-started>` tutorial.
+
+.. include:: /includes/subscriber-note.rst
+
+.. TODO:
+.. Change link in includes subscriber note to get started page instead of quick start primer
+
+The ``_id`` Field
+-----------------
+
+In a MongoDB collection, each document *must* contain an ``_id`` field
+with a unique field value.
+
+MongoDB allows you to manage this field in two ways:
+
+- You can set this field for each document yourself, ensuring each
+  ``_id`` field value is unique.
+- You can let the driver automatically generate unique ``ObjectId``
+  values for each document ``_id``. If you do not manually set an
+  ``_id`` value for a document, the driver populates the field
+  with an ``ObjectId``.
+
+Unless you can guarantee uniqueness, we recommend
+letting the driver automatically generate ``_id`` values.
+
+.. note::
+
+   Duplicate ``_id`` values violate unique index constraints, which
+   causes the driver to return a ``WriteError`` from
+   ``insertOne()`` or a ``BulkWriteError`` from ``insertMany()``.
+
+To learn more about the ``_id`` field, see the
+:manual:`Unique Indexes </core/index-unique/>` guide in the MongoDB Server manual.
+
+To learn more about document structure and rules, see the
+:manual:`Documents </core/document>` guide in the MongoDB Server manual.
+
+Insert One Document
+-------------------
+
+To add a single document to a MongoDB collection, call the ``insertOne()``
+method and pass the document you want to add along with the subscribe method
+
+The following example inserts a document into the ``restaurants`` collection:
+
+.. code-block:: java
+   :copyable: true
+
+    Document document = new Document("name", "Mongo's Burgers");
+    ObservableSubscriber<InsertOneResult> insertOneSubscriber = new SubscriberHelpers.OperationSubscriber<>();
+    restaurants.insertOne(document).subscribe(insertOneSubscriber);
+    insertOneSubscriber.await();
+
+Insert Multiple Documents
+-------------------------
+
+To add multiple documents to a MongoDB collection, call the ``insertMany()``
+method and pass a list of documents you want to add. Use 
+
+The following example inserts a list of documents into the ``restaurants`` collection:
+
+.. code-block:: java
+   :copyable: true
+
+    Document doc1 = new Document("name", "Mongo's Pizza");
+    Document doc2 = new Document("name", "Mongo's Coffee");
+
+    List<Document> documents = new ArrayList<Document>();
+
+    documents.add(doc1);
+    documents.add(doc2);
+
+    ObservableSubscriber<InsertManyResult> insertManySubscriber = new SubscriberHelpers.OperationSubscriber<>();
+    restaurants.insertMany(documents).subscribe(insertManySubscriber);
+    insertManySubscriber.await();
+
+Modify Insert Behavior
+----------------------
+
+To modify insert behavior, use the ``InsertOneOptions()`` and
+``InsertManyOptions()`` optional methods,
+which represent options you can use to configure the insert operation.
```

Comment:
S: These are [classes](https://mongodb.github.io/mongo-java-driver/5.1/apidocs/mongodb-driver-core/com/mongodb/client/model/InsertManyOptions.html), not methods

Feel free to edit this suggestion:

```suggestion
The ``InsertOneOptions`` class contains methods that modify
the behavior of the ``insertOne()`` method. To use the ``InsertOneOptions``
class, construct a new instance of the class, then call one or more of its methods
to customize the insert operation. You can chain these method calls together.
Pass the class instance and method calls as the second argument to the
``insertOne()`` method.

You can similarly modify the ``insertMany()`` method by using the
``InsertManyOptions()`` class.
```